### PR TITLE
Movie list related pull requests

### PR DESCRIPTION
**Pull request by Manny810**
These pull requests are specifically created in relation to week 2 with movie list related features.

### CODE REVIEW

File: app/src/main/java/com/example/whattowatch/NewMovieListActivity.java [hunk 1]
```diff
@@ -95,6 +95,10 @@ public void done(ParseException e) {
         rvMovieList.setLayoutManager(new LinearLayoutManager(this));
     }
 
+    /**
+     *  onActivityResult is the method that runs when an activity comes back after a previous activity has finished
+     *  We are overwriting this method to get the movie we selected in the FindMovieActivity and adding it to our recyclerView
+     */
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/NewMovieListActivity.java:101]: You don't need to explain onActivityResult, but rather what onActivityResult does in the context of this activity. In this case it adds a new movie and notifies (notifies who?)

File: app/src/main/java/com/example/whattowatch/FindMovieActivity.java [hunk 2]
```diff
@@ -63,7 +64,11 @@ protected void onCreate(Bundle savedInstanceState) {
         btnMovieSearch.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
+                // The code in here is run when the search button is pressed
+
                 String api_url = String.format(MOVIE_SEARCH_API_URL, etMovieSearch.getText());
+
+                // we make the movie database api call here to get movies associated to the keyword we searched
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/FindMovieActivity.java]: nice. You can just write "get movies associated to the keyword we searched"

File: app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return movieLists.size();
+    }
+
+    class ViewHolder extends RecyclerView.ViewHolder {
+
+        TextView tvTitle;
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java]: are these public or private by default?

File: app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.example.whattowatch.R;
+import com.example.whattowatch.models.Movie;
+import com.example.whattowatch.models.MovieList;
+
+import java.util.List;
+
+public class MovieListAdapter extends RecyclerView.Adapter<MovieListAdapter.ViewHolder>{
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java:27]: Add a comment up here saying what the adapter does in one sentence

File: app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java [hunk 5]
```diff
@@ -52,5 +68,37 @@ public void onClick(View view) {
...
+        // Set a layout Manager on the recycler view
+        rvMovieLists.setLayoutManager(new LinearLayoutManager(getContext()));
+
+    }
+
+    protected void queryLists() {
+        // Specify which class to query
+        ParseQuery<MovieList> query = ParseQuery.getQuery(MovieList.class);
+        query.include(MovieList.KEY_USER);
+        query.addDescendingOrder(MovieList.KEY_CREATED_AT);
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java:91]: nice use of globals :)

File: app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java [hunk 6]
```diff
@@ -52,5 +68,37 @@ public void onClick(View view) {
...
+        query.findInBackground(new FindCallback<MovieList>() {
+            @Override
+            public void done(List<MovieList> movieLists, ParseException e) {
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java]: you can override functions for a specific instance of a class? 😮

File: app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java [hunk 7]
```diff
@@ -49,19 +53,40 @@ public int getItemCount() {
...
         }
 
         public void bind(MovieList movieList) {
+            tvMovieListTitle.setText(movieList.getTitle());
+            tvMovieListSize.setText(String.format("Number of movies: %d", movieList.getMovies().length()));
+            tvMovieListCreatedAt.setText(String.format("Created %s", getRelativeTimeAgo(movieList.getCreatedAt().toString())));
 
         }
+
+        // getRelativeTimeAgo("Mon Apr 01 21:16:23 +0000 2014");
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java:77]: nice

File: app/src/main/java/com/example/whattowatch/adapters/RecommenderAdapter.java [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+    private Context context;
+    private List<Movie> movies;
+
+    public RecommenderAdapter(Context context, List<Movie> movies) {
+        this.context = context;
+        this.movies = movies;
+    }
+
+    @NonNull
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/adapters/RecommenderAdapter.java:30]: curious, what is this decorator for?

File: app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java [hunk 9]
```diff
@@ -5,16 +5,49 @@
...
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import okhttp3.Headers;
 
 
 public class RecommenderFragment extends Fragment {
+    public static final String TAG = "MovieListFragment";
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java]: what is this tag used for?

File: app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java [hunk 10]
```diff
@@ -107,15 +110,31 @@ public void done(List<MovieList> newMovieLists, ParseException e) {
...
+                                Log.e(TAG, "Hit Json Exception", ex);
+                            }
+
 
                         }
 
                         @Override
                         public void onFailure(int statusCode, Headers headers, String response, Throwable throwable) {
-
+                            Log.e(TAG, "onFailure called");
```
Comment (daniel-suissa) [inline app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java:133]: oh, got it! neat way to log!

File: app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java (hunk 6 above)
Comment (Manny810) [reply app/src/main/java/com/example/whattowatch/fragments/MovieListFragment.java]: yes, when defining a FindCallback, you have to override and define the done method, as that is typically instance specific

File: app/src/main/java/com/example/whattowatch/adapters/RecommenderAdapter.java (hunk 8 above)
Comment (Manny810) [reply app/src/main/java/com/example/whattowatch/adapters/RecommenderAdapter.java:30]: @NonNull basically checks to make sure that a certain parameter is not null, in this case we have to make sure that parent parameter is a true instance of ViewGroup and is not just null, so we have NonNull to do check

File: app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java (hunk 9 above)
Comment (Manny810) [reply app/src/main/java/com/example/whattowatch/fragments/RecommenderFragment.java]: That tag should be "RecommenderFragment". That code is outdated, as I changed it yesterday once I caught that. TAGs are used for logging. Whenever we call Log, which puts code into our logcat, we tag everything we input, so typically it is common to have a TAG variable for each java file so you can identify what file each line in the log cat comes from

File: app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java (hunk 3 above)
Comment (Manny810) [reply app/src/main/java/com/example/whattowatch/adapters/MovieListAdapter.java]: public

**Pull request merged**